### PR TITLE
added conda bin environment to scripts, removed scipts/legacy

### PR DESCRIPTION
**Pull request by karlnyr**
This PR:

### Fixes

- Functionality with the updated conda binaries

### Removed

- scripts/legacy

**How to prepare for test**:
- [ ] ssh to hasta
- [ ] install on stage:
`bash servers/resources/hasta.scilifelab.se/update-demux-stage.sh [THIS-BRANCH-NAME]`

**How to test**:
- [ ] login to ...
- [ ] do ...

**Expected test outcome**:
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: scripts/2500/checkfornewrun.bash [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #!/bin/bash
 
 shopt -s nullglob
+CONDA_BINARY_PATH='/home/proj/production/bin/miniconda3/envs/P_demux'
```
Comment (henrikstranneheim) [inline scripts/2500/checkfornewrun.bash:4]:
```suggestion
PROD_CONDA_DIR="/home/proj/production/bin/miniconda3"
CONDA_EXE="${PROD_CONDA_DIR}/bin/conda"
```

File: scripts/2500/checkfornewrun.bash [hunk 2]
```diff
@@ -18,7 +19,7 @@ for RUNDIR in ${INDIR}/*; do
                 if grep -qs ',ctmr,' ${RUNDIR}/SampleSheet.csv; then
                     echo [${NOW}] ${RUN} is CTMR - transmogrifying SampleSheet.csv
                     cp ${RUNDIR}/SampleSheet.csv ${RUNDIR}/SampleSheet.ctmr
-                    demux sheet demux -a miseq ${RUNDIR}/SampleSheet.ctmr > ${RUNDIR}/SampleSheet.csv
+                    conda run --name "${CONDA_BINARY_PATH}" demux sheet demux -a miseq ${RUNDIR}/SampleSheet.ctmr > ${RUNDIR}/SampleSheet.csv
```
Comment (henrikstranneheim) [inline scripts/2500/checkfornewrun.bash:22]:
```suggestion
                    ${CONDA_EXE} run --name P_demux demux sheet demux -a miseq ${RUNDIR}/SampleSheet.ctmr > ${RUNDIR}/SampleSheet.csv
```

File: scripts/2500/checkfornewrun.bash [hunk 3]
```diff
@@ -30,7 +31,7 @@ for RUNDIR in ${INDIR}/*; do
                 cp ${RUNDIR}/SampleSheet.csv ${RUNDIR}/Data/Intensities/BaseCalls/
             else
                 echo converting ${RUNDIR}/SampleSheet.csv
-                demux sheet convert ${RUNDIR}/SampleSheet.csv > ${RUNDIR}/SampleSheet.conv
+                conda run --name "${CONDA_BINARY_PATH}" demux sheet convert ${RUNDIR}/SampleSheet.csv > ${RUNDIR}/SampleSheet.conv
```
Comment (henrikstranneheim) [inline scripts/2500/checkfornewrun.bash:34]:
```suggestion
                ${CONDA_EXE} run --name P_demux demux sheet convert ${RUNDIR}/SampleSheet.csv > ${RUNDIR}/SampleSheet.conv
```

**Review by henrikstranneheim — COMMENTED** [review]
I think this should work

File: .github/workflows/installations.yml [hunk 4]
```diff
@@ -1,47 +1,39 @@
...
+            - name: Install pip
+              # Make sure that we have correct python version and pip version
+              # For details see .github/workflows/server_info.md
+              run: |
+                  python -m pip install --upgrade pip==18.1
+                  echo $PYTHON_VERSION
+                  pip --version
+              env:
+                  PYTHON_VERSION: ${{matrix.python-version}}
+                  OLD_VERSION: 3.6
```
Comment (Karl-Svard) [inline .github/workflows/installations.yml:32]:
```suggestion
```
Can't this line just be removed as we won't be using this version of python anymore (if I interpreted it correctly)?

File: scripts/hiseqx/xdemuxtiles.bash [hunk 5]
```diff
@@ -73,12 +84,12 @@ if [[ ! -e ${RUNDIR}/SampleSheet.csv ]]; then
     if [[ ${IS_DUAL} == '8' ]]; then
         DUALINDEX_PARAM='--dualindex'
     fi
-    log "demux sheet fetch -a wgs ${DUALINDEX_PARAM} ${FC} > ${RUNDIR}/SampleSheet.csv"
-    demux sheet fetch -a wgs ${DUALINDEX_PARAM} "${FC}" > "${RUNDIR}/SampleSheet.csv"
+    log "${CONDA_RUN_COMMAND}/demux sheet fetch -a wgs ${DUALINDEX_PARAM} ${FC} > ${RUNDIR}/SampleSheet.csv"
```
Comment (Karl-Svard) [inline scripts/hiseqx/xdemuxtiles.bash:87]: Where is `CONDA_RUN_COMMAND` fetched from, is it already pre-defined by conda or something?

File: scripts/iseq/checkfornewrun.bash [hunk 6]
```diff
@@ -54,8 +68,8 @@ for RUN_DIR in "${IN_DIR}"/*; do
             date +'%Y%m%d%H%M%S' > "${RUN_DIR}/demuxstarted.txt"
 
             if [[ ! -e ${RUN_DIR}/SampleSheet.csv ]]; then
-                log "demux sheet fetch --application iseq --pad --longest ${FC} > ${RUN_DIR}/SampleSheet.csv"
-                demux sheet fetch --application iseq --pad --longest "${FC}" > "${RUN_DIR}/SampleSheet.csv"
+                log "${CONDA_RUN_COMMAND}/ sheet fetch --application iseq --pad --longest ${FC} > ${RUN_DIR}/SampleSheet.csv"
```
Comment (Karl-Svard) [inline scripts/iseq/checkfornewrun.bash:71]: Same question as above.

**Review by Karl-Svard — APPROVED** [review]
Nice! I'm not a conda expert by any means, but looks alright to me. Just have a few questions I got while reading through the changes :)

File: scripts/hiseqx/xdemuxtiles.bash (hunk 5 above)
Comment (karlnyr) [reply scripts/hiseqx/xdemuxtiles.bash]: Nope, I just missed to change the log command 😆

**Comment by karlnyr** [comment]
Updated in prod:
image

**Pull request merged**